servers/rendering/instance_uniforms.h

```diff
@@ -0,0 +1,74 @@
+/**************************************************************************/
+/*  instance_uniforms.h                                                   */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef INSTANCE_UNIFORMS_H
+#define INSTANCE_UNIFORMS_H
+
+#include "core/templates/hash_map.h"
+#include "core/variant/variant.h"
+#include "servers/rendering/storage/material_storage.h"
+
+class InstanceUniforms {
+public:
+	void free(RID p_self);
+
+	void materials_start();
+	void materials_append(RID p_material);
+
+	// Assign location() to instance offset if materials_finish returns true.
+	[[nodiscard]] bool materials_finish(RID p_self);
```

Comment:
```suggestion
	bool materials_finish(RID p_self);
```
Style nitpick, we shouldn't include [[nodiscard]] unless it is needed. Otherwise you are making the code harder to read for no benefit